MapboxCoreNavigation/MMEEventsManager.swift

```diff
@@ -111,6 +116,13 @@ struct EventDetails {
             totalTimeInBackground += abs(session.lastTimeInBackground.timeIntervalSinceNow)
         }
         percentTimeInForeground = totalTimeInPortrait + totalTimeInLandscape == 0 ? 100 : Int((totalTimeInPortrait / (totalTimeInPortrait + totalTimeInLandscape) * 100))
+        currentLegIndex = routeController.routeProgress.legIndex
+        totalLegCount = routeController.routeProgress.route.legs.count
+        currentStepIndex = routeController.routeProgress.currentLegProgress.stepIndex
+        totalStepCount = routeController.routeProgress.currentLeg.steps.count
+        
+        let maneuver = routeController.routeProgress.currentLegProgress.upComingStep ?? routeController.routeProgress.currentLegProgress.currentStep
+        maneuverHash = "\(maneuver.names?.joined(separator: ",") ?? "")-\(maneuver.destinations?.joined(separator: ",") ?? "")-\(maneuver.codes?.joined(separator: ",") ?? "")-\(maneuver.initialHeading ?? 0)-\(maneuver.finalHeading ?? 0)"
```

Comment:
Don't worry about combining this yourself. We can keep these as separate fields and join later during analysis.

We can add these in a follow on PR if you want to keep this scoped to just the index additions.